### PR TITLE
Move config to settings; add working days and opening hours.

### PR DESCRIPTION
**Pull request by ne-ray**
Move some configs to settings, move attribute street to optional, add work days and time, add validate current datetime

### CODE REVIEW

**Comment by vojtasvoboda** [comment]
Could you please update your master branch from the original master branch? And then update this branch from the master? Thanks a lot

**Comment by ne-ray** [comment]
ok

**Comment by ne-ray** [comment]
ready

File: README.md [hunk 1]
```diff
@@ -64,7 +64,11 @@ $facade->getReservationsByInterval(\Carbon\Carbon $from, \Carbon\Carbon $to);
 $facade->isDateAvailable(\Carbon\Carbon $date);
 ```
 
-## Override configurations
+## Configuration
+
+You can find plugin configuration at the CMS backend. But you can also set some values at plugin's config file. Config values
```
Comment (ne-ray) [inline README.md]: I move only resarvation and date format to settings. Mail, random, and other don't move to Settings.

File: assets/js/reservationform.js [hunk 2]
```diff
@@ -47,32 +47,39 @@ function disableAllTimes() {
...
     var month = date.getMonth() + 1;
     month = (month < 10 ? '0' : '') + month;
     var selectedDate = day + '/' + month + '/' + date.getFullYear();
 
+    // get today date
     var dateNow = new Date();
     var dateNowFormat = (dateNow.getDate() < 10 ? '0' : '') + dateNow.getDate();
     dateNowFormat += '/' + (dateNow.getMonth() < 9 ? '0' : '') + (dateNow.getMonth() + 1);
-    dateNowFormat += '/' + date.getFullYear();
+    dateNowFormat += '/' + dateNow.getFullYear();
```
Comment (vojtasvoboda) [inline assets/js/reservationform.js]: @Ne-Ray please check this change, if am I right?

Comment (ne-ray) [reply assets/js/reservationform.js]: You are right.

Comment (vojtasvoboda) [reply assets/js/reservationform.js]: 👍  I'm in 85% of my review.

Comment (ne-ray) [reply assets/js/reservationform.js]: @vojtasvoboda You are fix second param convertTimeToArray().

**Comment by vojtasvoboda** [comment]
I've completed the review. Just two last things:

1) We had `2 hours` for setting reservation length and it was so powerful - you could set `2 hours` (for a table in a restauration), `1 week` (for a cottage in the mountains) or `25 minutes` (for psychologist consultation). I would like to keep this powerfulness, so we need to update code + add a better comment to backend area.

2) I'm not sure if we still need `formats_datetime` setting value. It is used only there: https://github.com/vojtasvoboda/oc-reservations-plugin/pull/20/files#diff-c6cb0be5f8dede9c79e55d2a181dfe10R273 so maybe we can remove this value and for `Validator::getDateTimeFormat()` just return `dateFormat + ' ' + timeFormat`?

What do you thing?

**Comment by ne-ray** [comment]
@vojtasvoboda
1) You are right, but input '1 week' don't translate to other languages and difficult validate. ( 1,5 week, 1.5 years). May be split attribute `interval` to `interval_value` (integer) and `interval_scale` (dropdown-list `hours`, `days`, `week`, `month`, `year`)
2) You are right.

**Comment by ne-ray** [comment]
@vojtasvoboda
1) if User input interval 1 day, then form don't have pikatime and pikadate close days.

File: assets/js/reservationform.js (hunk 2 above)
Comment (vojtasvoboda) [reply assets/js/reservationform.js]: @Ne-Ray Yes, that second parameter was unused, is it ok?

**Comment by vojtasvoboda** [comment]
Could you please try that `1 day` reservation interval again with the latest version of this branch? It works for me know.

File: assets/js/reservationform.js (hunk 2 above)
Comment (ne-ray) [reply assets/js/reservationform.js]: @vojtasvoboda yes. I forgot remove second parameter.

**Comment by vojtasvoboda** [comment]
I will be updating this Issue at Sunday, so if you want, feel free to send some updates directly to this branch.

**Comment by ne-ray** [comment]
Ok. I try this update.

File: facades/ReservationsFacade.php [hunk 3]
```diff
@@ -128,7 +128,7 @@ public function getReservations()
      */
     public function getActiveReservations()
     {
-        return $this->reservations->notCancelled()->get();
+        return $this->reservations->notCancelled()->currentDate()->get();
```
Comment (ne-ray) [inline facades/ReservationsFacade.php:131]: What do you thing about it?

**Comment by ne-ray** [comment]
"Could you please try that 1 day reservation interval again with the latest version of this branch? It works for me know.". It is work.

File: facades/ReservationsFacade.php (hunk 3 above)
Comment (vojtasvoboda) [reply facades/ReservationsFacade.php:131]: 👍

**Comment by vojtasvoboda** [comment]
Hi @Ne-Ray, I'm done with code review and updates from my side. Are you Ok with all changes? Can I merge it to the master and bump version up?

**Comment by vojtasvoboda** [comment]
Hi @Ne-Ray, I've added Mocha unit tests for that convert functions. Could you please test it? It can be done by:

```
npm install
npm test
```

in plugin's directory. Thanks a lot

**Comment by ne-ray** [comment]
> Hi @Ne-Ray, I'm done with code review and updates from my side. Are you Ok with all changes? Can I merge it to the master and bump version up?  

It is work!

> Hi @Ne-Ray, I've added Mocha unit tests for that convert functions. Could you please test it? ... in plugin's directory. Thanks a lot

Test success.

**Comment by vojtasvoboda** [comment]
Are we ready to merge this issue? How about your bug with timezone, is it ok?

**Comment by ne-ray** [comment]
@vojtasvoboda I'm ready. Timezone is not bug, if timezone config for backend user not equal config/app.php then backend convert time.

**Review by vojtasvoboda — APPROVED** [review]

**Comment by vojtasvoboda** [comment]
@Ne-Ray thanks for your work, I really appreciate it.

Here I'm sending you a special coupon for [Reservations Units](https://octobercms.com/plugin/vojtasvoboda-reservationsunits) plugin, so you can buy it just for $5 (which is the lowest price I can set for you):

Coupon: ILYA

Thanks
Vojta

**Comment by ne-ray** [comment]
@vojtasvoboda thanks you too.

**Pull request merged**